apps/dashboard/app/lib/smart_attributes/attributes/auto_output_directory.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module SmartAttributes
+  class AttributeFactory
+    # Build this attribute object. Must specify a valid directory in opts
+    #
+    # @param opts [Hash] attribute's options
+    # @return [Attributes::AutoOutputDirectory] the attribute object
+    def self.build_auto_output_directory(opts = {})
+      Attributes::AutoOutputDirectory.new('auto_output_directory', opts)
+    end
+  end
+
+  module Attributes
+    class AutoOutputDirectory < Attribute
+      # Value of auto_output_directory attribute
+      # Defaults to first script path in the project
+      # @return [String] attribute value
+      def value
+        opts[:value].presence
+      end
+
+      def widget
+        'text_field'
+      end
+
+      def label(*)
+        (opts[:label] || 'Output Directory').to_s
```

Comment:
The label here should change. Also likely should respond to internationalization.